doc/sphinx/reference/onedim/grid-refinement.md

```diff
@@ -0,0 +1,229 @@
+entire domain. If the ratio of the curvature between two adjacent grid points is greater
+than the user-specified value, then an additional grid point is inserted between the two
+points. This criterion is used to ensure that the solution is well-resolved in regions
+where the solution has steep gradients.
+
+The specific criteria that is being enforced is:
+
+$$
+  |\left( \frac{d x_n}{d z} \right)_{j+1} - \left(\frac{d x_n}{d z} \right)_j| <= \gamma_{max} (\displaystyle\max_j \frac{dx_n}{dz} - \displaystyle\min_j \frac{dx_n}{dz})
+$$
+
+Where $ \gamma_{max} $ is a number between 0 and 1 that is specified by the user, $ \frac{dx_n}{dz} $
+is the derivative of the solution component being considered, and $ \displaystyle\max_j \frac{dx_n}{dz} $ and $ \displaystyle\min_j \frac{dx_n}{dz} $ are the
+maximum and minimum values of the solution component derivative across the domain. Visually, these
+quantities are shown in the image below.
+
+```{figure} /_static/images/onedim/refinement_curve_metric_plot.svg
+:width: 75%
+:alt: Diagram of the curve refinement criterion.
+:align: center
+
+Diagram of the curve refinement criterion.
+```
+
+At each point in the domain, a ratio can be calculated for each component as:
+
+$$
+  \gamma_j =  \frac{|\left( \frac{d x_n}{d z} \right)_{j+1} - \left(\frac{d x_n}{d z} \right)_j|}{\displaystyle\max_j \frac{dx_n}{dz} - \displaystyle\min_j \frac{dx_n}{dz}}
+$$
+
+Points where $ \gamma_j $ is greater than $ \gamma_{max} $ are marked to have an additional
+grid point inserted to the right of them.
+
+The rule-of-thumb for setting $ \gamma_{max} $ is that it prevents any two adjacent points
+from having a change in the slope of a solution component that is greater than $ \gamma_{max} \cdot 100\% $ of
+the maximum change in that component's slope across the domain.
+
+Note: The slope is computed using a backward difference scheme and so the estimate of the
+derivative at a grid point depends on the solution at the grid point and the solution at the
+grid point to the right.
+
+$$
+  \left( \frac{d x_n}{d z} \right)_j \approx \frac{x_{n,j+1} - x_{n,j}}{z_{j+1} - z_{j}}
+$$
+
+If a point is marked as needing an additional grid point, the refinement algorithm will
+actually insert a point to the right of the point under consideration as well as to the
+point that is to the right. This essentially splits the two intervals that are to the right
+of the point under consideration.
+
+
+### Prune
+
+This refinement option works in contrast to the other refinement options. Instead of
+adding grid points, this option removes grid points that are not needed. This option
+lets a user specify a value of the ratio quantities described above that if the ratio
+drops below this, then the point is removed. The user specifies a pruning value, $ \delta $,
+and that serves as a lower-bound on the ratio quantities.
+
+$$
+  \delta = \beta_{min} = \gamma_{min}
+$$
+
+If $\beta_j < \delta$, or $\gamma_j < \delta$, then the point is removed.
```

Comment:
I don't think this is correct -- the `prune` value is relative to either the `curve` or `slope` parameters, so the condition under which a point is allowed to be removed is like $\beta_j / \beta_{max} < \delta$. Also, the condition must be satisfied for _both_ `curve` and `slope`, otherwise successive refinement would potentially just re-add the point.

Reply:
Maybe I should say "is marked for removal" instead of "is removed"?